public/app/features/dashboard-scene/edit-pane/shared.tsx

```diff
@@ -0,0 +1,112 @@
+import { useSessionStorage } from 'react-use';
+
+import { SceneGridRow, SceneObject, VizPanel } from '@grafana/scenes';
+import { Dropdown, Button, IconButton, Stack, Icon, Menu } from '@grafana/ui';
+import { t } from 'app/core/internationalization';
+
+import { DashboardScene } from '../scene/DashboardScene';
+import { SceneGridRowEditableElement } from '../scene/layout-default/SceneGridRowEditableElement';
+import { EditableDashboardElement, isEditableDashboardElement } from '../scene/types/EditableDashboardElement';
+
+import { DashboardEditableElement } from './DashboardEditableElement';
+import { VizPanelEditableElement } from './VizPanelEditableElement';
+
+export function useEditPaneCollapsed() {
+  return useSessionStorage('grafana.dashboards.edit-pane.isCollapsed', false);
+}
+
+interface RenderTitleProps {
+  title: string;
+  onDelete?: () => void;
+  onCopy?: () => void;
+  onDuplicate?: () => void;
+}
+
+export const renderTitle = ({ title, onDelete, onCopy, onDuplicate }: RenderTitleProps) => {
```

Comment:
I think this is a stand-alone component as opposed to just a function. I suggest naming/using it like so and also extract it in its own file along with `MenuItems` as it would be easier to maintain on the long run.